src/datadog.ts

```diff
@@ -78,9 +86,14 @@ export class Datadog extends cdk.Construct {
         this.props.extensionLayerVersion,
       );
       redirectHandlers(lambdaFunctions, addLayers);
+
       if (this.props.forwarderArn !== undefined) {
-        log.debug(`Adding log subscriptions using provided Forwarder ARN: ${this.props.forwarderArn}`);
-        addForwarder(this.scope, lambdaFunctions, this.props.forwarderArn);
+        if (this.props.extensionLayerVersion !== undefined) {
+          log.debug(`Skipping adding subscriptions to the lambda functions since the extension is enabled`);
```

Comment:
Nit: I think the subscriptions get added to the lambda log groups and not directly to the function. 
Maybe this is clearer:
`Skipping adding subscriptions to the lambda log groups since the extension is enabled`